src/newViews/PillarAnalysis/EntriesFilterForm/index.tsx

```diff
@@ -139,6 +142,13 @@ function EntriesFilterForm(props: OwnProps) {
         setValue(filtersValue ?? initialValue);
     }, [filtersValue, setValue]);
 
+    useEffect(() => {
+        setFieldValue(selectedNgram, 'search');
+    }, [
+        selectedNgram,
+        setFieldValue,
+    ]);
```

Comment:
Instead of modifying the form, why not update the actual filter?

Reply:
ok